CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 # Copyright (c) Microsoft Corporation.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-cmake_minimum_required(VERSION 3.15)
+cmake_minimum_required(VERSION 3.17)

```

Comment:
I don't think we can do this without also changing the instructions for building from the IDE in the README. Visual Studio 2019 16.6p2 currently ships with CMake 3.16.5 and so those build instructions would be broken by this change.

Reply:
Good point. I'll roll back the cmake edits.

Reply:
I added "Ensure dependent tools like CMake and LLVM match those shipping with the equivalently supported Visual Studio release." to https://github.com/microsoft/STL/wiki/Checklist-for-Updating-Toolset-and-or-Dependencies